pkg/execute/executor.go

```diff
@@ -11,6 +11,7 @@ import (
 	"gopkg.in/yaml.v3"
 
```

Comment:
Is it ready to review? I'm not sure if I'm testing that correctly, but when I run on this branch:
```bash
helm install botkube --namespace botkube ./helm/botkube -f /tmp/values.yaml --wait
```

Where `/tmp/values.yaml` is:
```
communications:
  'default-group':
    # Settings for Slack
    slack:
      token: {TOKEN}
```

And `values.yaml` has the following diff (you can save it to file and apply with `git apply`):

```yaml
diff --git a/helm/botkube/values.yaml b/helm/botkube/values.yaml
index 6e1d063..f7336d9 100644
--- a/helm/botkube/values.yaml
+++ b/helm/botkube/values.yaml
@@ -13,11 +13,11 @@ image:
   # -- BotKube container image registry.
   registry: ghcr.io
   # -- BotKube container image repository.
-  repository: kubeshop/botkube
+  repository: kubeshop/pr/botkube
   # -- BotKube container image pull policy.
-  pullPolicy: IfNotPresent
+  pullPolicy: Always
   # -- BotKube container image tag. Default tag is `appVersion` from Chart.yaml.
-  tag: v9.99.9-dev
+  tag: 677-PR
 
 # -- Configures Pod Security Policy to allow BotKube to run in restricted clusters.
 # [Ref doc](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
@@ -284,11 +284,35 @@ executors:
         # Example: `ignore: ["x", "y", "secret-ns-*"]`
         ignore: []
       # -- If true, enables `kubectl` commands execution.
-      enabled: false
+      enabled: true
+      ## List of allowed `kubectl` commands.
+      commands:
+        # -- Configures which `kubectl` methods are allowed.
+        verbs: ["describe", "diff", "explain", "get", "logs", "top", "auth"]
+        # -- Configures which K8s resource are allowed.
+        resources: ["deployments", "pods", "namespaces", "daemonsets", "statefulsets", "storageclasses", "nodes", "configmaps"]
+      # -- Configures the default Namespace for executing BotKube `kubectl` commands. If not set, uses the 'default'.
+      defaultNamespace: default
+      # -- If true, enables commands execution from configured channel only.
+      restrictAccess: false
+
+  'kubectl-read-only2':
+    ## Kubectl executor configuration.
+    kubectl:
+      namespaces:
+        # -- List of allowed Kubernetes Namespaces for command execution. The "all" keyword, allows all the Namespaces.
+        include:
+          - all
+        # -- List of ignored Kubernetes Namespace. It can be used only when `include: ["all"]`.
+        # It can contain a wildcard (*) that expands to zero or more arbitrary characters
+        # Example: `ignore: ["x", "y", "secret-ns-*"]`
+        ignore: []
+      # -- If true, enables `kubectl` commands execution.
+      enabled: true
       ## List of allowed `kubectl` commands.
       commands:
         # -- Configures which `kubectl` methods are allowed.
-        verbs: ["api-resources", "api-versions", "cluster-info", "describe", "diff", "explain", "get", "logs", "top", "auth"]
+        verbs: ["api-resources", "api-versions", "cluster-info"]
         # -- Configures which K8s resource are allowed.
         resources: ["deployments", "pods", "namespaces", "daemonsets", "statefulsets", "storageclasses", "nodes", "configmaps"]
       # -- Configures the default Namespace for executing BotKube `kubectl` commands. If not set, uses the 'default'.
@@ -314,14 +338,20 @@ communications:
     ## Settings for Slack.
     slack:
       # -- If true, enables Slack bot.
-      enabled: false
+      enabled: true
       # -- Map of configured channels. The `channels` property name is an alias for a given configuration.
       #
       ## Format: channels.<alias>
       channels:
         'default':
-          # -- Slack channel name without '#' prefix where you have added BotKube and want to receive notifications in.
-          name: 'SLACK_CHANNEL'
+          name: botkube-test
+          bindings:
+            executors:
+              - kubectl-read-only
+            sources:
+              - k8s-events
+        'two':
+          name: general
           bindings:
             executors:
               - kubectl-read-only
@@ -450,7 +480,7 @@ communications:
 ## Global BotKube configuration.
 settings:
   # -- Cluster name to differentiate incoming messages.
-  clusterName: not-configured
+  clusterName: dev
   # -- If true, restarts the BotKube Pod on config changes.
   configWatcher: true
   # -- If true, notifies about new BotKube releases.
```

I still get all enabled executors in a given installation, even if `kubectl-read-only2` is not bound to the channel:
```
Enabled executors:
  kubectl:
    kubectl-read-only:
      namespaces:
        include:
          - all
      enabled: true
      commands:
        verbs:
          - describe
          - diff
          - explain
          - get
          - logs
          - top
          - auth
        resources:
          - deployments
          - pods
          - namespaces
          - daemonsets
          - statefulsets
          - storageclasses
          - nodes
          - configmaps
      defaultNamespace: default
      restrictAccess: false
    kubectl-read-only2:
      namespaces:
        include:
          - all
      enabled: true
      commands:
        verbs:
          - api-resources
          - api-versions
          - cluster-info
        resources:
          - deployments
          - pods
          - namespaces
          - daemonsets
          - statefulsets
          - storageclasses
          - nodes
          - configmaps
      defaultNamespace: default
      restrictAccess: false
```

Am I testing it wrong?